### PR TITLE
Cleanup crypter interface

### PR DESCRIPTION
**Pull request by philip-cc**
### Description
This PR cleans up the crypter interface / `crypto.go`:
* Remove the unnecessary nested interface. 
* Remove unnecessary nested `AESGCMEncrypt`/`AESGCMDecrypt`.
* Rename `crypt.go` to `aes.go`.
* Move password stuff out of `aes.go`.

### Parent Issue
Issue #130 

### Related Pull Requests
None

### Comments for the Reviewers
None

### Type(s) of Change (Split the PR if you check many)
- [ ] Added user feature
- [ ] Added technical feature
- [ ] Added tests
- [x] Refactor
- [ ] Bugfix
- [ ] Updated documentation
- [ ] Infrastructure changes (Terraform, GCP, K8s, other)
- [ ] CI/CD workflow changes

### Testing Checklist
- [ ] I have added/updated unit tests for functional changes.
- [ ] I have added/updated integration tests for changes that affect dependent systems.
- [ ] I have added/updated end-to-end tests for feature changes.

### General Checklist
- [x] I have pulled in the latest changes from master.
- [x] I have run `make lint`.
- [x] I have successfully run `make docker-up; make tests`.
- [ ] I have updated relevant workflows and deployment tools.
- [ ] I have updated/added relevant documentation (readme, doc comments, etc).
- [ ] I have added the license to new source code files.
- [x] I have spent some time looking over the full diff before creating this PR.

### CODE REVIEW

File: encryption-service/impl/crypt/aes.go [hunk 1]
```diff
@@ -132,11 +60,19 @@ func (c *AESCrypter) Decrypt(ciphertext, aad, key []byte) ([]byte, error) {
...
 	if err != nil {
 		return nil, err
 	}
 
+	aesgcm, err := cipher.NewGCM(aesblock)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = aesgcm.Open(data[:0], nonce, data, aad)
```
Comment (bruno-cc) [inline encryption-service/impl/crypt/aes.go:75]: I would still do a `if err != nil` here, as it's a very dangerous spot and I really never want to return invalid stuff here if any error happens

**Review by bruno-cc — COMMENTED** [review]
See minor comment while we are at this spot.

File: encryption-service/impl/crypt/aes.go (hunk 1 above)
Comment (philip-cc) [reply encryption-service/impl/crypt/aes.go:75]: Missed that somehow, fixed!

**Review by bruno-cc — APPROVED** [review]
Thanks, looks good.

**Review by bruno-cc — APPROVED** [review]

**Pull request merged**